dora/src/vm/class.rs

```diff
@@ -76,6 +76,10 @@ pub struct Class {
     // true if this class is the generic Array class
     pub is_array: bool,
     pub is_str: bool,
+    // temporary measure as classes are never zero-initializable – but as Int, Long, Float, etc. are
+    // currently "declared" as classes, not structs, so we use this field until it's possible to
+    // migrate them to structs.
+    pub is_zero_initializable: bool,
```

Comment:
I don't think we need this. `cls.ty` will give you the BuiltinType for a class, if this is `BuiltinType::Class(_)` then this is a *real* class. Int/Long/Float/etc. will have BuiltinType::Int/Long/Float/etc. instead.

Reply:
This would also mean that you don't need to change anything in prelude.rs (apart from initializing zero_trait_id).

Reply:
Ok, will change it.

Reply:
(THe idea was that this gave us some more flexibility for the case where e. g. `Option`'s `none` case was compiled down to zeroes, but the definition wasn't changed from class to struct...

Reply:
Maybe we don't need the flag in those cases as well. We might be able to just check the memory layout of the ADT and don't need a separate flag.

Reply:
I assumed it would make sense if we didn't recompute this information every time we need it, but only computed it once per class and saved it there.